src/battle_ai_main.c

```diff
@@ -2710,6 +2711,14 @@ static s32 AI_DoubleBattle(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
               || gMovesInfo[aiData->partnerMove].criticalHitStage > 0
               || HasMoveWithCriticalHitChance(battlerAtkPartner))
             ADJUST_SCORE(GOOD_EFFECT);
+        break;
+    case EFFECT_SPICY_EXTRACT:
+        if (HasMoveWithCategory(battlerAtkPartner, DAMAGE_CATEGORY_PHYSICAL)
```

Comment:
Would be good to also make sure this will have a good effect by checking:
- Partner's Attack can be raised
- Partner didn't intentionally try to boost their Defense
- Partner doesn't have Contrary or Good as Gold

Reply:
I added `AI_ShouldSpicyExtract` that (hopefully) checks all of it.